frontend/src/Components/OAuth2ScopesComponent.tsx

```diff
@@ -0,0 +1,36 @@
+import styles from './OAuth2ScopesComponent.module.scss';
+import { OAuth2ScopesLabels } from '../Types/OAuth2';
+
+function ScopeControl(props: { name: string, description: string, checked: boolean }) {
+  return (
+    <label>
+      <input
+        disabled={true}
+        type={'checkbox'} {...(props.checked ? { defaultChecked: true } : {})}
+        name={props.name}/>
+      <span>
+        <b>{props.name}</b> - {props.description}
+      </span>
+
+    </label>
+  );
+}
+
+export default function OAuth2ScopesComponent(props: { appRequests: string | null }) {
+  const requestedScopesNames = ['openid'].concat(props.appRequests?.split(' ').map(s => s.trim()) || []);
+  return (<div className={styles.container}>
+    {
+      Object.keys(OAuth2ScopesLabels)
```

Comment:
ok, this is super dangerous

this view shows only the subset of the scopes that are hardcoded on the client

the scopes that are actually requested, and will be saved to db are not validated

Reply:
fixed, scopes are filtered on both the backend and the frontend